src/main.c

```diff
@@ -178,6 +179,53 @@ static bool BarMainGetStations (BarApp_t *app) {
 	return ret;
 }
 
+static bool BarMainGetUserProfile(BarApp_t *app) {
+	PianoReturn_t pRet;
+	CURLcode wRet;
+	bool ret;
+
+	BarUiMsg (&app->settings, MSG_INFO, "Get user profile ... ");
+	ret = BarUiPianoCall (app, PIANO_REQUEST_GET_USER_PROFILE, NULL, &pRet, &wRet);
+	return ret;
+}
+
+static bool BarMainGetAllStations (BarApp_t *app) {
+	PianoReturn_t pRet;
+	CURLcode wRet;
+	bool ret;
+	PianoRequestDataGetPlaylist_t reqData;
+	reqData.station = app->nextStation;
+	reqData.quality = app->settings.audioQuality;
+	reqData.retPlaylist = NULL;
+
+	do {
+		ret = BarMainGetStations (app);
+		if(!ret) {
+			break;
+		}
+		BarUiMsg (&app->settings, MSG_INFO, "Get items ... ");
+		ret = BarUiPianoCall (app, PIANO_REQUEST_GET_ITEMS, &reqData, &pRet, &wRet);
+		if(!ret) {
+			break;
+		}
+		BarUiMsg (&app->settings, MSG_INFO, "Annotate Objects ... ");
+		ret = BarUiPianoCall (app, PIANO_REQUEST_ANNOTATE_OBJECTS, &reqData, &pRet, &wRet);
+		if(!ret) {
+			break;
+		}
+		if(app->ph.user.IsPremiumUser) {
```

Comment:
I’d remove these kind of checks unless the API calls always return an error for non-premium users (apparently they don’t).